miqi/runtime/turn_runner.py

```
@@ -632,6 +824,131 @@ async def run_agent_job(self, job: Any) -> TurnResult:
+            "permissions": permissions_for_tools(tool_names),
+            "timeout_seconds": 300,
+        })
+        answers = result.get("answers") or {}
+        return bool(
+            result.get("status") == "submitted"
+            and str(answers.get("choice_id", "")) == "confirm"
+        )
+
+
+    async def _emit_todo_state(self, turn: Any) -> None:
+        """v3.3 Step 4（后端）：TodoState 变更推前端（display=todo_state）。
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)